src/zenml/zen_stores/base_zen_store.py

```diff
@@ -309,37 +299,19 @@ def get_default_store_config(path: str) -> StoreConfiguration:
 
     def _initialize_database(self) -> None:
         """Initialize the database on first use."""
-        try:
-            default_workspace = self._default_workspace
-        except KeyError:
-            default_workspace = self._create_default_workspace()
-        try:
-            assert self._admin_role
-        except KeyError:
-            self._create_admin_role()
-        try:
-            assert self._guest_role
-        except KeyError:
-            self._create_guest_role()
+        default_workspace = self._get_or_create_default_workspace()
 
         config = ServerConfiguration.get_server_config()
-        # If the auth scheme is external, don't create the default user and
-        # stack
+        # If the auth scheme is external, don't create the default user
         if config.auth_scheme != AuthScheme.EXTERNAL:
             try:
-                default_user = self._default_user
-            except KeyError:
-                default_user = self._create_default_user()
-            try:
-                self._get_default_stack(
-                    workspace_name_or_id=default_workspace.id,
-                    user_name_or_id=default_user.id,
-                )
+                _ = self._default_user
             except KeyError:
-                self._create_default_stack(
-                    workspace_name_or_id=default_workspace.id,
-                    user_name_or_id=default_user.id,
-                )
+                self._create_default_user()
+
+        self._get_or_create_default_stack(
+            workspace=default_workspace,
+        )
```

Comment:
I guess this is true for the default stack belonging to the default workspace, but this doesn't apply to other workspaces. Yes, we don't officially allow/support multiple workspaces ATM, but I can see why it can't hurt to create those default stacks in the migration script.

Reply:
Yep exactly, if someone for some reason has created a different workspace they also get a default stack during the migration. Creating a new workspace however will not create a default stack, which is why I'm also not sure we should keep the logic in the migration here.

Reply:
I actually had to change this, creating a new workspace now automatically creates a default stack for that workspace. We still need it in the migration, as default stack creation only happens when the workspace is created and not for existing ones.

Reply:
I saw that, good idea !